### PR TITLE
Add JMoney API with additional TWR and lambda helper APIs

### PR DESCRIPTION
**Pull request by HaloFour**
Adds the JMoney Java API bridge back to Money 0.9 with additional helper APIs supporting try-with-resources and lambdas.

### CODE REVIEW

File: money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+
+import com.comcast.money.api.Note;
+import com.comcast.money.core.Money;
+import com.comcast.money.core.Tracer;
+
+/**
+ * Java API for working directly with money from Java code
+ */
+public class JMoney {
+    private static Tracer tracer;
```
Comment (pauljamescleary) [inline money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java]: this is never set (unless I missed it)?

Should always use `Money.Environment().tracer()`

File: money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+    }
+
+    /**
+     * @return the {@link com.comcast.money.core.Tracer} to use for tracing
+     */
+    public static Tracer tracer() {
+
+        Tracer current = tracer;
+        if (current == null) {
+            return Money.Environment().tracer();
```
Comment (pauljamescleary) [inline money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java]: I would always return `Money.Environment().tracer()`, this way the `tracer` method becomes convenience.

File: money-core/src/test/scala/com/comcast/money/core/japi/JMoneySpec.scala [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+      val captor = ArgumentCaptor.forClass(classOf[Note[_]])
+      verify(tracer).record(captor.capture())
+      val note = captor.getValue
+      val nil: AnyRef = null
+      note.name shouldBe "nill"
+      note.value shouldBe nil
+    }
+  }
+}
```
Comment (pauljamescleary) [inline money-core/src/test/scala/com/comcast/money/core/japi/JMoneySpec.scala]: ahh, new line!

File: money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java (hunk 2 above)
Comment (HaloFour) [reply money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java]: I'm only allowing `Tracer` to be set in order to support the unit tests.  Do you have a preferred pattern to follow?  This one is similar to what `SpringMoney` does.

File: money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java (hunk 1 above)
Comment (HaloFour) [reply money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java]: There is a `protected void setTracer` method, used by the unit tests to pass a mock `Tracer`

**Comment by HaloFour** [comment]
I've also been toying with an async version of the API which would pop the span and only stop it when the returned future completes.  It involves a bit more plumbing and would want to review the concept before I'd suggest adding it, perhaps in a future PR.

Here's a gist of the implementation: https://gist.github.com/HaloFour/8c55faef7135355ae31a3ebabf2dd9e3

File: money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java (hunk 2 above)
Comment (pauljamescleary) [reply money-core/src/main/scala/com/comcast/money/core/japi/JMoney.java]: Is it possible to allow unit tests the ability to set the static tracer, and just initialize it by default to `Money.Environment().tracer()`?

**Pull request merged**